pkg/geo/geomfn/segmentize_test.go

```diff
@@ -0,0 +1,186 @@
+		},
+		{
+			wkt:              "POLYGON ((0.0 0.0, 1.0 0.0, 1.0 1.0, 0.0 0.0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.1))",
+			maxSegmentLength: 5,
+			expectedWKT:      "POLYGON ((0.0 0.0, 1.0 0.0, 1.0 1.0, 0.0 0.0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.1))",
+		},
+		{
+			wkt:              "POLYGON EMPTY",
+			maxSegmentLength: 1,
+			expectedWKT:      "POLYGON EMPTY",
+		},
+		{
+			wkt:              "MULTIPOINT ((1.0 1.0), (2.0 2.0))",
+			maxSegmentLength: 1,
+			expectedWKT:      "MULTIPOINT ((1.0 1.0), (2.0 2.0))",
+		},
+		{
+			wkt:              "MULTILINESTRING ((1.0 1.0, 2.0 2.0, 3.0 3.0), (6.0 6.0, 7.0 6.0))",
+			maxSegmentLength: 1,
+			expectedWKT:      "MULTILINESTRING ((1.0 1.0, 1.5 1.5, 2.0 2.0, 2.5 2.5, 3.0 3.0), (6.0 6.0, 7.0 6.0))",
+		},
+		{
+			wkt:              "MULTILINESTRING (EMPTY, (1.0 1.0, 2.0 2.0, 3.0 3.0), (6.0 6.0, 7.0 6.0))",
+			maxSegmentLength: 1,
+			expectedWKT:      "MULTILINESTRING (EMPTY, (1.0 1.0, 1.5 1.5, 2.0 2.0, 2.5 2.5, 3.0 3.0), (6.0 6.0, 7.0 6.0))",
+		},
+		{
+			wkt:              "MULTIPOLYGON (((3.0 3.0, 4.0 3.0, 4.0 4.0, 3.0 3.0)), ((0.0 0.0, 1.0 0.0, 1.0 1.0, 0.0 0.0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.1)))",
+			maxSegmentLength: 1,
+			expectedWKT:      "MULTIPOLYGON (((3.0 3.0, 4.0 3.0, 4.0 4.0, 3.5 3.5, 3.0 3.0)), ((0.0 0.0, 1.0 0.0, 1.0 1.0, 0.5 0.5, 0.0 0.0), (0.1 0.1, 0.2 0.1, 0.2 0.2, 0.1 0.1)))",
+		},
+		{
+			wkt:              "GEOMETRYCOLLECTION (POINT (40.0 10.0), LINESTRING (10.0 10.0, 20.0 20.0, 10.0 40.0), POLYGON ((40.0 40.0, 20.0 45.0, 45.0 30.0, 40.0 40.0)))",
+			maxSegmentLength: 10,
+			expectedWKT:      "GEOMETRYCOLLECTION (POINT (40.0 10.0), LINESTRING (10.0 10.0, 15.0 15.0, 20.0 20.0, 16.666666666666668 26.666666666666668, 13.333333333333334 33.33333333333333, 10.0 40.0), POLYGON ((40.0 40.0, 33.333333333333336 41.66666666666667, 26.666666666666668 43.333333333333336, 20.0 45.0, 28.333333333333336 40.0, 36.66666666666667 35.0, 45.0 30.0, 42.5 35.0, 40.0 40.0)))",
+		},
+		{
+			wkt:              "MULTIPOINT ((0.0 0.0), (1.0 1.0))",
+			maxSegmentLength: -1,
+			expectedWKT:      "MULTIPOINT ((0.0 0.0), (1.0 1.0))",
+		},
+	}
+	for _, test := range segmentizeTestCases {
+		t.Run(fmt.Sprintf("%s, maximum segment length: %f", test.wkt, test.maxSegmentLength), func(t *testing.T) {
+			geom, err := geo.ParseGeometry(test.wkt)
+			require.NoError(t, err)
+			modifiedGeom, err := Segmentize(geom, test.maxSegmentLength)
+			require.NoError(t, err)
+			expectedGeom, err := geo.ParseGeometry(test.expectedWKT)
+			require.NoError(t, err)
+			require.Equal(t, expectedGeom, modifiedGeom)
+		})
+	}
+	// Test for segment maximum length as negative for geometry collection.
+	t.Run(fmt.Sprintf("%s, maximum segment length: %f", segmentizeTestCases[13].wkt, 0.0), func(t *testing.T) {
```

Comment:
same comment about minting a new `wkt` here.

Reply:
done!
Same done for `geography` case.